security-jwt/src/main/java/io/micronaut/security/token/jwt/signature/jwks/JwksSignature.java

```diff
@@ -47,17 +53,16 @@
  * @since 1.1.0
  */
 @EachBean(JwksSignatureConfiguration.class)
-public class JwksSignature implements SignatureConfiguration {
+public class JwksSignature implements JwksCache, SignatureConfiguration {
 
+    @Deprecated
     public static final int DEFAULT_REFRESH_JWKS_ATTEMPTS = 1;
 
     private static final Logger LOG = LoggerFactory.getLogger(JwksSignature.class);
-
     private final JwkValidator jwkValidator;
-
+    private final JwksSignatureConfiguration jwksSignatureConfiguration;
+    private Date jwkSetCachedAt;
     private JWKSet jwkSet;
```

Comment:
I still think this needs a `volatile` to be correct.

Reply:
I've added `volatile` to both.